commerce-cloud-code/app_yourshophere/cartridge/partials/tile/tile.js

```diff
@@ -1,86 +1,20 @@
+ * @returns {Object} The view model containing the model
  */
-exports.createModel = () => {
-    const HttpSearchParams = require('*/api/URLSearchParams');
-    const httpParams = new HttpSearchParams(request.httpParameterMap);
-    const componentSettings = require('*/cartridge/utils/ComponentSettings').get(httpParams.get('component'));
-
-    const tileSearch = require('*/api/ProductSearchModel').get(httpParams, { swatchAttribute: componentSettings.swatchDimension });
-    tileSearch.search();
-
-    const imageFilter = {
-        key: componentSettings.swatchDimension,
-        value: httpParams.get('color'),
-    };
-
-    const hit = tileSearch.foundProducts[0];
-    const model = {};
-
-    if (hit) {
-        model.name = name.createModel(hit);
-        model.image = image.createModel({
-            hit,
-            search: tileSearch,
-            imageFilter,
-            config: {
-                imageViewType: componentSettings.imageViewType,
-                imageDISConfig: componentSettings.imageDISConfig,
-            },
-        });
-        model.price = price.createModel({
-            hit,
-            tileSearch,
-            httpParams,
-        });
-        model.swatches = swatches.createModel({
-            hit,
-            search: tileSearch,
-            config: { swatchAttribute: componentSettings.swatchDimension },
-        });
-        model.addToCartButton = require('./addtocartbutton').createModel(hit);
-        model.analytics = JSON.stringify({
-            type: 'productView',
-            id: hit.object.productID,
-        });
-        model.analyticsContribution = JSON.stringify({
-            contributesTo: [
-                'viewCategory',
-                'viewSearch',
-            ],
-            contributionOptions: { property: 'products', mode: 'array-push' },
-            value: {
-                id: hit.object.productID,
-                sku: '',
-                altId: '',
-                altIdType: '',
-            },
-        });
-    }
-
-    return model;
+exports.createModel = function createModel(input) {
+    return input.model;
 };
 
 /**
- * Render the product tile
- * @param {Object} model - The model for the product tile
- * @returns {string} The HTML for the product tile
+ * Render the tile
+ * @param {Object} model - The view model containing the model
+ * @returns {string} The HTML template for the tile
  */
 exports.template = (model) => `
 <article data-include-url="${request.httpQueryString}" data-analytics='${model.analytics}' data-analytics-contribution='${model.analyticsContribution}'>
-    <header>${name.template(model.name)}</header>
-    <body>
-        ${image.template(model.image)}
-        ${price.template(model.price)}
-        ${swatches.template(model.swatches)}
-    </body>
-    <footer>${addToCartButton.template(model.addToCartButton)}</footer>
+    ${model.regions.tileHeader.setTagName('header').setComponentTagName('custom-deleteme').render()}
```

Comment:
As commented above, I'd make this a single region, and then have
```
model.regions.tileHeader.setTagName('article').render()
```

Components could then maybe have their own custom tags like header or footer.

Reply:
but we have multiple in the body, and I don't like if the component prescribe their position.

The "name" doesn't have or rather must not hardcode that it is the "header"